File: examples/update_status.py
#!/usr/bin/python3

import pika
import json

url = 'amqps://USER:PASS@rabbit.suse.de?heartbeat_interval=5';
connection = pika.BlockingConnection(pika.URLParameters(url))
channel = connection.channel()
channel.exchange_declare(exchange='pubsub', exchange_type='topic',
                         passive=True, durable=True)
message = json.dumps({'name': 'test', 'status': 'ok'})
channel.basic_publish(exchange='pubsub', routing_key='suse.bot.status',
                      body=message)
connection.close()

File: examples/update_status.pl
#!/usr/bin/env perl
use Mojo::Base -strict;

use Mojo::RabbitMQ::Client;

my $url = 'amqps://USER:PASS@rabbit.suse.de:5671?exchange=pubsub';
my $publisher = Mojo::RabbitMQ::Client->publisher(url => $url);
$publisher->publish_p({name => 'test', status => 'ok'},
  routing_key => 'suse.bot.status')->then(sub { say 'Status updated' })
  ->catch(sub { say "Updating status failed: $_[1]" })->wait;
